src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+
+    public OrderController(OrderService orderService,
+                           DtoResponseMapper<OrderResponseDto,
+                                   Order> orderDtoResponseMapper,
+                           UserService userService,
+                           ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
+        return orderService.getOrdersHistory(userService.get(userId)).stream()
+                .map(orderDtoResponseMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userid) {
+        return orderDtoResponseMapper
+               .toDto(orderService
+                        .completeOrder(shoppingCartService
+                               .getByUser(userService
+                                       .get(userid))));
```

Comment:
Such way is much better because you have reduced the number of cascades in your code
```suggestion
Order order = orderService.completeOrder(shoppingCartService
	.getByUser(userService.get(userid)))
return orderDtoResponseMapper.toDto();
```